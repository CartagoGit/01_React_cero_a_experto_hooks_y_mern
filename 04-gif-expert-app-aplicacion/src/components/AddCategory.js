import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ categories="", setCategories }) => {
	const [inputValue , setInputValue] = useState(categories);
	// console.log(inputValue);

	const handleInputChange = (e) => {
		console.log("-> handleInputChange");
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		console.log("-> handleSubmit",inputValue);
		e.preventDefault();

		if (inputValue.trim().length > 0) {
			// setCategories((c) => [inputValue, ...c]);
			setCategories(() => inputValue);
			setInputValue("");
		} else setInputValue("");
	};

	return (
		<form onSubmit={handleSubmit}>
			<input type='text' value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
};
