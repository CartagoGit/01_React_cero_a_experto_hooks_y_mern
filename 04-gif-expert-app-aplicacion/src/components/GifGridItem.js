import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    // console.log(id,title,url);
    return (
        <div className='card animate__tada' id={id}>
            {/* {props} */}
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
