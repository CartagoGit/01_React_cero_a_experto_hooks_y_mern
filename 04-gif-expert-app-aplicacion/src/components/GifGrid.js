// import React, { useState, useEffect } from "react";
import React from "react";
import { useFetchGif } from "../hooks/useFetchGifs";
// import { getGifs } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
	// const [count, setCount] = useState(0);
	// const [images, setImages] = useState([]);

	// useEffect(() => {
	// 	getGifs(category).then((imgs) => setImages(imgs));
	// 	//// eslint-disable-next-line react-hooks/exhaustive-deps
	// }, [category]);

	const { data: images, loading } = useFetchGif(category);

	return (
		<>
			<h3>{category}</h3>
            {loading && <p>Cargando...</p>}

			

			<div className='cardGrid'>
				{/* { <h3>{count}</h3> } */}
				{/* <button onClick={ () => setCount(count+1)}></button> */}

				{images.map((img) => (
					// <li key={id}> {title}</li>
					// <GifGridItem key={img.id} img ={img}/>
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};
