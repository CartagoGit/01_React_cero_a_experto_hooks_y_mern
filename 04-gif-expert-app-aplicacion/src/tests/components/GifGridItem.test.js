import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas en el componente <GifGridItem />>", () => {
	let wrapper;
	let data = { id: 0, title: "", url: "" };
	const fillWrapper = (
		id = 0,
		title = "un titulo",
		url = "https://unaurl.com"
	) => {
		data = { id, title, url };
		wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);
	};

	beforeEach(() => fillWrapper());

	test("Debe mostrar el componente correctamente ", () => {
		fillWrapper(123, "jei", "http");
		// console.log(1, wrapper.html());
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de tener un parrafo con el texto del titulo", () => {
		// console.log(2, wrapper.html());
		const p = wrapper.find("p").text().trim();
		expect(p).toBe(data.title);
	});

	test('Debe tener la imagen = al url y alt de los props y el id en el contenedor', () => {
		const img = wrapper.find('img').at(0);
		const card = wrapper.find('.card').at(0);
		// console.log(img.props());
		expect(img.prop('src')).toBe(data.url);
		expect(img.prop('alt')).toBe(data.title);
		// console.log(card.props().id);
		expect(card.props().id).toBe(data.id);
	});

	test('Debe tener animate__tada', () => {
		const clase = wrapper.find('div').at(0);
		const search = 'animate__tada';
		const className= clase.props().className;
		expect(className.includes(search)).toBe(true);
		expect(className).toContain(search);
	});
	
	
});
