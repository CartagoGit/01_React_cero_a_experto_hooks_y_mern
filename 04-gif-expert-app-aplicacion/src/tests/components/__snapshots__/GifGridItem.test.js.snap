// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem />> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__tada"
  id={123}
>
  <img
    alt="jei"
    src="http"
  />
  <p>
    jei
  </p>
</div>
`;
